src/Marten/Events/Projections/ProjectionOptions.cs

```diff
@@ -190,17 +190,36 @@ internal IProjection[] BuildInlineProjections(DocumentStore store)
     /// <param name="lifecycle">Override the snapshot lifecycle. The default is Inline</param>
     /// <param name="asyncConfiguration">
     ///     Optional configuration including teardown instructions for the usage of this
-    ///     projection within the async projection daempon
+    ///     projection within the async projection daemon
+    /// </param>
+    /// <returns>The extended storage configuration for document T</returns>
+    public MartenRegistry.DocumentMappingExpression<T> Snapshot<T>(
+        SnapshotLifecycle lifecycle,
+        Action<AsyncOptions> asyncConfiguration = null
+    ) =>
+        singleStreamProjection<T>(lifecycle.Map(), null, asyncConfiguration);
+
+    /// <summary>
+    /// Perform automated snapshot on each event for selected entity type
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="lifecycle">Override the snapshot lifecycle. The default is Inline</param>
+    /// <param name="configureProjection">Use it to further customize the projection.</param>
+    /// <param name="asyncConfiguration">
+    ///     Optional configuration including teardown instructions for the usage of this
+    ///     projection within the async projection daemon
     /// </param>
     /// <returns>The extended storage configuration for document T</returns>
     public MartenRegistry.DocumentMappingExpression<T> Snapshot<T>(
         SnapshotLifecycle lifecycle,
+        Action<SingleStreamProjection<T>> configureProjection,
         Action<AsyncOptions> asyncConfiguration = null
     ) =>
-        singleStreamProjection<T>(lifecycle.Map(), asyncConfiguration);
+        singleStreamProjection<T>(lifecycle.Map(), configureProjection, asyncConfiguration);
```

Comment:
@oskardudycz I made an overload instead of adding an optional parameter to the existing method, like you said.

But this way, the overload does not receive the same amount of coverage like the existing one (unless I copy all  the test). 

If you are fine with this I leave it as is.
Otherwise, I would prefer to have it as an optional parameter on the existing method.

Reply:
You have to make it an overload because of semantic version rules. The extra optional parameter would force users to recompile. Ticky tacky rule here, but the overload is necessary.

Reply:
So then it's just whether to copy some tests to get similar coverage or leave as is?

Reply:
Copying tests sounds fine to me. You could also add an integration test checking snapshot deletion for inline snapshot.

Reply:
@jeremydmiller I added a test that is equal to my actual use case. _(and I did not copy all other tests as discussed with you on discord)_